### PR TITLE
Zeebe User Tasks

### PR DESCRIPTION
**Pull request by pinussilvestrus**
* Adds support to model user tasks in Zeebe diagrams
* Adds the Headers and Input/Output tab for user tasks
* Adds the new user task form definition for user tasks

Related to camunda/camunda-modeler#2104

Bildschirmfoto 2021-02-23 um 08 47 06

Can be tested via

```sh
npx @bpmn-io/sr camunda/camunda-bpmn-js#2104-zeebe-user-tasks -c "npm run start:cloud"
```

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
@andreasgeier fyi: I think a design review is not necessary for this one, since it includes one input field only. The rest is part of what we discussed yesterday.

**Comment by pinussilvestrus** [comment]
The zeebe-bpmn-moddle changes are released and integrated ✅

File: lib/camunda-cloud/features/properties-provider/parts/HeadersProps.js [hunk 1]
```diff
@@ -38,3 +38,13 @@ export default function(group, element, bpmnFactory, translate) {
...
+
+
+// helpers ////////////////
+
+function canHaveHeaders(element) {
+  return isAny(element, [
+    'bpmn:ServiceTask',
+    'bpmn:UserTask'
+  ]);
+}
```
Comment (barmac) [inline lib/camunda-cloud/features/properties-provider/parts/HeadersProps.js:50]: I believe this is a clear sign that we need a `bpmn-js` extension in `zeebe-bpmn-moddle`. With the current state, I can morph a User Task to a Receive Task and `zeebe:taskHeaders` stay in the extension elements.

Comment (pinussilvestrus) [reply lib/camunda-cloud/features/properties-provider/parts/HeadersProps.js:50]: Good catch, but this still exists outside of this PR, right? E.g. By morphing a Service Task to a Receive Task.

Shall we create an issue on `zeebe-bpmn-moddle`?

Comment (pinussilvestrus) [reply lib/camunda-cloud/features/properties-provider/parts/HeadersProps.js:50]: Seems like we already [have this extension](https://github.com/zeebe-io/zeebe-bpmn-moddle/blob/master/lib/extension.js#L62), but no [`allowedIn` meta for headers](https://github.com/zeebe-io/zeebe-bpmn-moddle/blob/master/resources/zeebe.json#L111).

Comment (barmac) [reply lib/camunda-cloud/features/properties-provider/parts/HeadersProps.js:50]: Oh, I missed that 🤪  Let's have a separate issue then. I'll create it.

Comment (barmac) [reply lib/camunda-cloud/features/properties-provider/parts/HeadersProps.js:50]: [Done](https://github.com/zeebe-io/zeebe-bpmn-moddle/issues/7)

File: lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            rootElement = getRootElement(element),
+            body = values.formJSON;
+
+      let commands = [],
+          userTaskForm,
+          formId;
+
+      // (0) cleanup user task form + form definition on empty body
+      if (!body || body === '') {
+        return cleanupUserTaskForm(element);
```
Comment (barmac) [inline lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js:93]: We don't perform the cleanup when a User Task is morphed into another type or when it's removed. The result is that we end up with redundant extension elements on the process.

Comment (pinussilvestrus) [reply lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js:93]: an alternative could be to handle it via a modeling behavior.

**Review by barmac — COMMENTED** [review]
I left one comment to be addressed and also have two remarks below:

* How about if we validate the form configuration to not accept non-JSON input?
* A single-line input looks a bit odd for a multi-line JSON. Still, if this is a transition state I am OK with leaving this as it is.

File: lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js (hunk 2 above)
Comment (barmac) [reply lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js:93]: That would be a perfect place for this 🏅

**Comment by barmac** [comment]
Great job overall :)

**Comment by pinussilvestrus** [comment]
> * How about if we validate the form configuration to not accept non-JSON input?
> * A single-line input looks a bit odd for a multi-line JSON. Still, if this is a transition state I am OK with leaving this as it is.

Good points. I think since we have a transition state here, it's okay to not have validation and the single line input. I'd not put any more effort into the user interaction.

File: lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js (hunk 2 above)
Comment (pinussilvestrus) [reply lib/camunda-cloud/features/properties-provider/parts/implementation/FormDefinition.js:93]: I'll tackle this

**Comment by pinussilvestrus** [comment]
I added another commit (bb9697351c037d9916724d5a4456d515eb7be44c) which adds an modeling behavior which

* ensures cleanup on deletion & replace
* ensures a new user task form with the same content on copy & paste

Ready for review again.

**Comment by pinussilvestrus** [comment]
@barmac @MaxTru any chance to get this thing reviewed / merged?

**Comment by barmac** [comment]
I can look into that today.

**Comment by barmac** [comment]
One unexpected behaviour:

When I copy a User Task from one diagram to another,
the form definition as well as `zeebe:userTaskForm` are copied
but the form contents are removed (Form JSON Configuration is empty).

I am OK with merging the PR as it is.

**Comment by pinussilvestrus** [comment]
I think that's pretty much because we can't currently copy the userTaskForm properly, since we don't know where to get the value from, since the `rootElement` from the copied one isn't available in the newer diagram. 

I would also be fine with merging this and wait once someone is reporting it.

**Review by barmac — APPROVED** [review]

**Pull request merged**